### PR TITLE
Prevent invalid cover art hash values

### PR DESCRIPTION
**Pull request by uklotzde**
https://mixxx.discourse.group/t/mixxx-2-2-4-reads-id3v1-tags-mixxx-2-3-or-2-4-does-not/21041/11

### CODE REVIEW

File: src/library/coverart.cpp [hunk 1]
```diff
@@ -51,14 +51,22 @@ QString coverInfoToString(const CoverInfo& info) {
...
 #if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
             image.sizeInBytes()
 #else
             image.byteCount()
 #endif
-            );
+    );
+    // In rare cases the calculated checksum could be equal to the
+    // reserved value defaultHash() which might cause unexpected
+    // behavior In this case we simply invert all bits to get a hash
```
Comment (Be-ing) [inline src/library/coverart.cpp:64]:
```suggestion
    // behavior. In this case we simply invert all bits to get a hash
```

**Comment by uklotzde** [comment]
The renaming and namespacing in main will require some manual work when merging 2.3.

**Comment by uklotzde** [comment]
CI failure on Windows in unrelated (WiX)

**Comment by Holzhaus** [comment]
> CI failure on Windows in unrelated (WiX)

How did that break?

**Comment by uklotzde** [comment]
> > CI failure on Windows in unrelated (WiX)
> 
> How did that break?

IDK. Doesn't matter for this quick fix.

**Review by Holzhaus — APPROVED** [review]
Fix looks good, thanks. Didn't test this locally yet.

**Review by daschuer — APPROVED** [review]
LGTM, thank you.

**Comment by uklotzde** [comment]
Merging to main...

**Pull request merged**